docs/references/ui-tests.md

```diff
@@ -0,0 +1,67 @@
+# UI Tests
+
+UI test are used to prevent visual regressions. When a PR is put up, the CI pipeline runs a series of UI tests and takes visual snapshots inside each test. Each snapshot is compared to the baseline snapshot to check for any changes. If there are changes the snapshots must be updated as part of the PR. UI tests use [Playwright](https://playwright.dev/docs/intro) to generate snapshots.
+
+UI tests must only be updated as part of the CI pipeline, this is because the build agents may run on a different OS and have different graphics setup that your local machine (so there will be slight pixel differences between locally generated snapshots and ones generated by the CI pipeline).
+
+## Updating UI tests in the CI pipeline
+
+Add `ui-change` label to the PR. The CI pipeline will now automatically update all snapshots in your PR.
+
+## Running UI Tests locally
+
+Normally you do not have to run the UI tests locally, but for debugging CI failures it can be necessary. To run locally:
+
+```sh
+# This example will build calling react-composite tests.
+
+# Navigate to packlet directory
+cd packages/react-composites
+
+# Build the e2e tests
+rushx build:e2e:call
+
+# Set a connection string environment variable
+$env:CONNECTION_STRING = '<connection string>' # This differs per development environment, this command is for powershell
```

Comment:
Maybe we should have how to do this step for in code spaces with Bash since a number of members on the team or working that way? I will look into it if you agree that is a good idea!

Reply:
That would be great!